_posts/2018-08-01-umbrellas-just-when-it-rains.md

```diff
@@ -0,0 +1,84 @@
+
+## Getting to know Umbrella Applications
+
+If you haven't already, take a peek at the [Umbrella](/en/lessons/advanced/umbrella-projects/) lesson which provides a great overview of Umbrella applications.
+
+## Separating concerns
+
+In my humble opinion one of the great strengths of Umbrellas is the forced separation of concerns and decoupling of code.
+We have to be explicit with our individual application's configuration and this includes dependencies on other peer applications.
+If our presentation layer doesn't have the internal data application as a dependency there's no chance those two components will become coupled.
+
+While the same separation can be achieved in theory using a singular Application, the Umbrella gives us the added benefit of enforcing it through dependencies.
+I've never seen "in theory" stand-up in a professional setting when skill levels and experience vary, Umbrellas remove any questions and confusion.
+
+Above any other reason, this is often the deciding factor for whether or not I leverage an Umbrella application.
+
+## Service Oriented Architect lite
+
+If we know we are building a complex application with internal components that will grow in size with distinct roles and responsibilities, Umbrellas can be a valuable tool in the toolchest.
+
+With one codebase and repository we're able to maintain any number of sub applications as part of our overall Umbrella.
+If we're building an online marketplace we could use this to keep our payment transaction code isolated from our other modules.
+We can implement integrations with third-party services as standalone applications inside our Umbrella.
+This not only keeps those concerns separated but allows us to work on components individually and more importantly: test them individually.
+
+Nothing is keeping us from running multiple Phoenix applications in a single Umbrella.
+We can develop our admin web portal independently of our customer facing site, running on a separate port making it a matter of network configuration to limit access to VPN users only.
+
+If the scope of an application grows too much that we need to remove it from the Umbrella, no problem!
+Doing this is simple with Umbrellas because of those explicit dependencies, updating those references is all we need to do.
+
+An added benefit of developing our project as disparate micro-services: large teams can work concurrently on one codebase while minimizing code conflicts.
+
+## One deployment to rule them all
+
+What's easier: deploying changes to 12 applications or 1?
+
+Releases aren't just easier with Umbrellas, they're more powerful.
+We not only have a single artifact we can deploy, we can configura
+
+## To Umbrella or not to Umbrella
+
+Contrary to what some may believe, Umbrella applications aren't a perfect solution for every problem.
+Before we `mix new` we should pause and consider some simple rules to help guide our decision making process:
```

Comment:
it might be worth referencing the automatically created supervision tree here. It may seem implicit to someone experienced but maybe someone who is not could think of an umbrella as capable of hosting multiple `mix new appname` vs `mix new appname --sup` which is delegated to how you create it in execution but in the sense of composing pieces of logic if you arent familiar with OTP it might seem reasonable to batch together some libraries this way

Reply:
@notactuallypagemcconnell I'm not 💯 I follow, could you give me a snippet of what you might suggest?